osu.Game/Skinning/LegacySkinExtensions.cs

```diff
@@ -63,8 +64,51 @@ IEnumerable<Texture> getTextures()
             }
         }
 
-        public static bool HasFont(this ISkin source, string fontPrefix)
-            => source.GetTexture($"{fontPrefix}-0") != null;
+        public static bool HasFont(this ISkin source, LegacyFont font)
+        {
+            return source.GetTexture($"{source.GetFontPrefix(font)}-0") != null;
+        }
+
+        public static string GetFontPrefix(this ISkin source, LegacyFont font)
+        {
+            switch (font)
+            {
+                case LegacyFont.Score:
+                    return source.GetConfig<LegacySetting, string>(LegacySetting.ScorePrefix)?.Value ?? "score";
+
+                case LegacyFont.Combo:
+                    return source.GetConfig<LegacySetting, string>(LegacySetting.ComboPrefix)?.Value ?? "score";
+
+                case LegacyFont.HitCircle:
+                    return source.GetConfig<LegacySetting, string>(LegacySetting.HitCirclePrefix)?.Value ?? "default";
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(font));
+            }
+        }
+
+        /// <summary>
+        /// Returns the numeric overlap of number sprites to use.
+        /// A positive number will bring the number sprites closer together, while a negative number
+        /// will split them apart more.
+        /// </summary>
+        public static float GetFontOverlap(this ISkin source, LegacyFont font)
+        {
+            switch (font)
+            {
+                case LegacyFont.Score:
+                    return source.GetConfig<LegacySetting, float>(LegacySetting.ScoreOverlap)?.Value ?? -2f;
+
+                case LegacyFont.Combo:
+                    return source.GetConfig<LegacySetting, float>(LegacySetting.ComboOverlap)?.Value ?? -2f;
```

Comment:
As an aside I don't know that these defaults are correct. I know those were the values before, and that [wiki states it's -2](https://osu.ppy.sh/wiki/en/Skinning/skin.ini), but I think it might actually be 0.

`HitCircleOverlap` does seem to have a default of -2, though.

Reply:
Yeah well, all I have as a reference would be the osu! wiki 😛

Reply:
Indeed these are 0, and as @bdach mentioned, `HitCircleOverlap` is -2.